clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java

```diff
@@ -471,22 +452,22 @@ public boolean rename(Path src, Path dst) throws IOException {
 
     TachyonURI srcPath = new TachyonURI(Utils.getPathWithoutScheme(src));
     TachyonURI dstPath = new TachyonURI(Utils.getPathWithoutScheme(dst));
-    TachyonFile srcFile = tryOpen(srcPath);
-    FileInfo info;
+    ensureExists(srcPath);
+    URIStatus dstStatus;
     try {
-      TachyonFile file = mTFS.open(dstPath);
-      info = mTFS.getInfo(file);
+      dstStatus = mTFS.getStatus(dstPath);
     } catch (IOException e) {
-      info = null;
+      dstStatus = null;
     } catch (TachyonException e) {
-      info = null;
+      dstStatus = null;
     }
     // If the destination is an existing folder, try to move the src into the folder
-    if (info != null && info.isFolder) {
+    if (dstStatus != null && dstStatus.isFolder()) {
       dstPath = dstPath.join(srcPath.getName());
     }
     try {
-      return mTFS.rename(srcFile, dstPath);
+      mTFS.rename(srcPath, dstPath);
+      return true;
```

Comment:
I don't think the return has changed here, since we always caught the exception and returned false.